quickwit-search/src/root.rs

```diff
@@ -110,21 +113,24 @@ pub struct ErrorRetries {
 /// - Is the node ok, but just the split failing?
 /// - Did all requests fail? (Should we retry in that case?)
 fn analyze_errors(search_result: &SearchResultsByAddr) -> Option<ErrorRetries> {
-    let contains_errors = search_result.values().any(|res| {
-        res.is_err()
-            || res
-                .as_ref()
-                .map(|ok_res| ok_res.failed_requests.is_empty())
-                .unwrap_or(false)
+    let contains_retryable_errors = search_result.values().any(|res| {
+        res.as_ref().map_or(true, |ok_res| {
```

Comment:
nitpick: It is probably just me but I have trouble reading `map_or` code.
Also I don't see how it helps compared to `.map(...).or(...)`

In this specific case, I would prefer a 


```
if let Some(leaf_search_result) = leaf_search_result_opt {
    contains_retryable_error(leaf_search_result)
} else {
    true
}
```

Reply:
I think map_or is a little hard to read because the parameter order is or and then map.

I adopted your proposal. They are both weird to read because of the error double layering and think they should be flattened.